bluesky-social/social-app
# Movable following feed

Backend PR: https://github.com/bluesky-social/atproto/pull/2427

**To do**
- [x] Integrate new post-create-account handling, setting default feeds

### Description

This PR adds the ability to move your following feed and even un-pin it. To do so, it runs a migration on first load from v1 saved feeds prefs to v2. See linked PR above.

V2 preferences have different interfaces for reading and writing this data. So this PR touches all of those usages in order to make this work. I'll leave comments in the code with more details.

Testing:
- create prod test account with a few feeds, saved and/or pinned, on bsky.app
  - log in to branch deploy https://social-app-pr-3593.onrender.com/
  - should migrate smoothly, play around
- un-pin all feeds
- on ProfileFeed/List screens, test removing/pinning in all configurations, Feeds and SavedFeeds should stay in sync
  - note: you can remove saved lists now!
- note that your test profile on bsky.app is still fine and usable, but that changes to feeds aren't reflected on the branch deploy
- create an account (can't do from branch deploy)
  - create a new account, select some algos, including Discover and complete onboarding, feeds should look as expected
  - create another account, select NO algos, feeds should be Timeline/Discover

![CleanShot 2024-04-24 at 12 47 48@2x](https://github.com/bluesky-social/social-app/assets/4732330/0eedda9e-44f6-43e3-809d-1987e5f7f854)
![CleanShot 2024-04-24 at 12 48 10@2x](https://github.com/bluesky-social/social-app/assets/4732330/f05dc255-85c7-4f40-a34f-315478bbb455)
![CleanShot 2024-04-24 at 12 48 12@2x](https://github.com/bluesky-social/social-app/assets/4732330/0e3204c9-5f52-4497-87d5-65aeb6b52f1c)
![CleanShot 2024-04-24 at 12 48 21@2x](https://github.com/bluesky-social/social-app/assets/4732330/0c5cd383-1863-4620-940b-4053d85f0bbd)
![CleanShot 2024-04-24 at 12 48 25@2x](https://github.com/bluesky-social/social-app/assets/4732330/80217fea-deb1-40b9-aa51-49543a8c2413)
![CleanShot 2024-04-24 at 12 50 55@2x](https://github.com/bluesky-social/social-app/assets/4732330/be7bb359-48a8-4cee-b2ce-3c8157227f24)
